fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -209,6 +210,24 @@ public void prefixPrimaryKeysWithNullByteAfterPrefix(int formatVersion, boolean
         }
     }
 
+    @Test
+    public void testLoadStoreHeader() throws Exception {
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            // load store header with new static method
+            RecordMetaDataProto.DataStoreInfo dataStoreInfo1 = FDBRecordStore.loadStoreHeaderAsync(context, path).get();
+            // load store header with old method
+            RecordMetaDataProto.DataStoreInfo dataStoreInfo2 = recordStore.getRecordStoreState().getStoreHeader();
+            // assert
```

Comment:
`RecordMetaDataProto.DataStoreInfo` is a protobuf so it would be better to just assert that the two `datastoreInfo`s are equals, so that this test maintains coverage if those protobuf definitions add fields.